pkg/controller/egress/controller.go

```diff
@@ -449,3 +458,75 @@ func (c *EgressController) enqueueEgresses(poolName string) {
 		c.queue.Add(egress.Name)
 	}
 }
+
+func (c *EgressController) updateEgressAllocatedCondition(egress *egressv1beta1.Egress, err error) {
+	var desiredCondition *egressv1beta1.EgressCondition
+	if egress.Spec.ExternalIPPool != "" {
+		if err == nil {
+			desiredCondition = &egressv1beta1.EgressCondition{
+				Type:               egressv1beta1.IPAllocated,
+				Status:             v1.ConditionTrue,
+				Reason:             "Allocated",
+				Message:            "EgressIP is successfully allocated",
+				LastTransitionTime: metav1.Now(),
+			}
+		} else {
+			desiredCondition = &egressv1beta1.EgressCondition{
+				Type:               egressv1beta1.IPAllocated,
+				Status:             v1.ConditionFalse,
+				Reason:             "AllocationError",
+				Message:            fmt.Sprintf("Cannot allocate EgressIP from ExternalIPPool %s due to: %v", egress.Spec.ExternalIPPool, err),
+				LastTransitionTime: metav1.Now(),
+			}
+		}
+	}
+
+	toUpdate := egress.DeepCopy()
+	var updateErr, getErr error
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		actualCondition := getIPAllocatedCondition(toUpdate)
+		if compareConditionIgnoringTimestamp(actualCondition, desiredCondition) {
+			return nil
+		}
+		var newConditions []egressv1beta1.EgressCondition
+		for _, c := range toUpdate.Status.Conditions {
+			if c.Type != egressv1beta1.IPAllocated {
+				newConditions = append(newConditions, c)
+			}
+		}
+		if desiredCondition != nil {
+			newConditions = append(newConditions, *desiredCondition)
+		}
+		toUpdate.Status.Conditions = newConditions
+		_, updateErr = c.crdClient.CrdV1beta1().Egresses().UpdateStatus(context.TODO(), toUpdate, metav1.UpdateOptions{})
+		if updateErr != nil && errors.IsConflict(updateErr) {
+			if toUpdate, getErr = c.crdClient.CrdV1beta1().Egresses().Get(context.TODO(), egress.Name, metav1.GetOptions{}); getErr != nil {
+				return getErr
+			}
+		}
+		return updateErr
+	}); err != nil {
+		klog.ErrorS(err, "Error updating Egress Status")
+	}
+}
+
+// getIPAllocatedCondition gets the IPAllocated condition from an Egress
+func getIPAllocatedCondition(egress *egressv1beta1.Egress) *egressv1beta1.EgressCondition {
+	for _, c := range egress.Status.Conditions {
+		if c.Type == egressv1beta1.IPAllocated {
+			return &c
+		}
+	}
+	return nil
+}
+
+// compareConditionIgnoringTimestamp compares two conditions ignoring the timestamp
+func compareConditionIgnoringTimestamp(condition1, condition2 *egressv1beta1.EgressCondition) bool {
+	if condition1 == nil || condition2 == nil {
+		return false
+	}
+	if condition1 == nil && condition2 == nil {
+		return true
+	}
```

Comment:
It would return wrong result with this order, it should be:
```
	if condition1 == nil && condition2 == nil {
		return true
	}
	if condition1 == nil || condition2 == nil {
		return false
	}
```